listener.go

```
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"log"
+
+	"github.com/JSainsburyPLC/ingestion-sdk/bus"
+	"github.com/JSainsburyPLC/ingestion-validation-service/configuration"
+)
+
+func listen(c *configuration.Configuration) bus.Delivery {
+	session, err := bus.NewSession(c.RabbitMQ.Dial)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	queueOptions := bus.QueueOptions{
+		Name:         bus.ValidatorQueueName,
+		Durable:      false,
+		DeleteUnused: false,
+		Exclusive:    false,
+		NoWait:       false,
+		Args:         nil,
+	}
+	q, _ := session.DeclareQueue(queueOptions)
+
+	bindOptions := bus.BindOptions{
+		Name:     q.Name,
+		Key:      bus.ValidatorQueueKey,
+		Exchange: bus.ValidatorQueueExchange,
+		NoWait:   false,
+		Args:     nil,
+	}
+	err = session.DeclareBind(bindOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	consumeOptions := bus.ConsumeOptions{
+		Queue:     q.Name,
+		Consumer:  "",
+		AutoACK:   true,
+		Exclusive: false,
+		NoLocal:   false,
+		NoWait:    false,
+		Args:      nil,
+	}
+
+	delivery, err := session.Consume(consumeOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
```

could we return an error?